src/ctap/mod.rs

```diff
@@ -345,27 +382,23 @@ where
                         Duration::from_ms(TOUCH_TIMEOUT_MS),
                     );
                 }
-                match (&command, &self.stateful_command_type) {
-                    (
-                        Command::AuthenticatorGetNextAssertion,
-                        Some(StatefulCommand::GetAssertion(_)),
-                    ) => (),
-                    (
+                match (&command, self.stateful_command_permission.get_command()) {
```

Comment:
Should such logic actually be part of `StatefulPermission`?
e.g. in `check_command_permission`

Reply:
I'd need different `check_command` commands, one for the `command` here, the other for the `subcommand` in credential management. Since these `match`es were specific to the call site, I thought I was best to keep them there.
Another problem is that merging with `check_command_permission` means we start returning a `Result` here, and this caller function does not return an error, but converts to the byte later. It might mean that we have to refactor this `process_command` function.
WDYT?

Reply:
Ok.
I thought that the logic was "generic" in the way that if you start iterating through the credentials, we know which set of commands are allowed without dropping the stat. So at the same time you could have checked the command and the timestamp. Any invalid pair would result in invalidating the state.
And the rest of the code would only have to care about the requested command, not the internal state command.

Reply:
I'll resolve since you're okay, just a comment on idea to conserve my thoughts. I think doing one check for everything would be nice, but there are a few problems in that idea:

- The timeout check has to be made periodically even when no commands arrive, so we don't miss when our timestamps overflow.
- The subcommands are parsed later than the command byte, and we need both for the decision. If CTAP2 wouldn't have so many nested commands, parsing would be easier and cleaner.
- We'd need to refactor `process_command` a bit.

I'd rather keep these matches where the commands are parsed, so our struct doesn't need any knowledge of other commands or subcommands.